### PR TITLE
Add "without" option to proxy middleware

### PR DESCRIPTION
**Pull request by brk0v**
For https://github.com/mholt/caddy/issues/95

Caddyfile:
```
proxy /api localhost:9999 {
    without /api
}
```

Request ```/api/messages``` transformed to ```/messages``` and send to  localhost:9999.

### CODE REVIEW

File: middleware/proxy/reverseproxy.go [hunk 1]
```diff
@@ -73,6 +75,9 @@ func NewSingleHostReverseProxy(target *url.URL) *ReverseProxy {
 		} else {
 			req.URL.RawQuery = targetQuery + "&" + req.URL.RawQuery
 		}
+		if without != "" {
+			req.URL.Path = strings.Replace(req.URL.Path, without, "", 1)
```
Comment (mholt) [inline middleware/proxy/reverseproxy.go]: strings.TrimPrefix might be better here.

File: middleware/proxy/upstream.go [hunk 2]
```diff
@@ -28,13 +28,13 @@ type staticUpstream struct {
 		Path     string
 		Interval time.Duration
 	}
+	Without string
```
Comment (mholt) [inline middleware/proxy/upstream.go]: The name "Without" in code is a little nebulous. Maybe something more descriptive like TrimPrefix or StripPrefix or WithoutPathPrefix or something like that. (Thoughts?)

**Comment by mholt** [comment]
Thanks for working on this! Looks like the tests fail now, since the function signature changed. Can you fix the tests? Also, if you could add at least one test for your change, that would be appreciated.

**Comment by brk0v** [comment]
I've done code fixes. 
About tests. Stuck with them. Wrote one, but it was strange and overkill from my point of view.

**Comment by mholt** [comment]
Great - thanks Viacheslav! This will be very useful.

**Pull request merged**